packages/gatsby/src/utils/develop-proxy.ts

```diff
@@ -0,0 +1,72 @@
+import { createServer } from "http"
+import httpProxy from "http-proxy"
+import fs from "fs-extra"
+import { getServices } from "gatsby-core-utils/dist/service-lock"
+import restartingScreen from "./restarting-screen"
+
+interface IProxyControls {
+  serveRestartingScreen: () => void
+  serveSite: () => void
+  server: any
+}
+
+const noop = (): void => {}
+
+export const startDevelopProxy = (input: {
+  proxyPort: number
+  targetPort: number
+  programPath: string
+}): IProxyControls => {
+  let shouldServeRestartingScreen = false
+
+  const proxy = httpProxy.createProxyServer({
```

Comment:
Do we need a proxy here or can we just serve something on a different port since this is specific to gatsby admin?

Reply:
This is not specific to Gatsby Admin at all, in fact this whole patch is useful on its own! This proxy has to run on the actual user port so we can serve the "Gatsby is restarting..." screen while the develop process is restarting.

Reply:
Oh right! That makes sense.